mesonbuild/dependencies/base.py

```diff
@@ -355,6 +355,13 @@ def generate_link_whole_dependency(self) -> Dependency:
         new_dep.libraries = []
         return new_dep
 
+    def get_as_static(self, recursive: bool) -> Dependency:
+        new_dep = copy.copy(self)
+        new_dep.libraries = [getattr(lib, 'static', lib) for lib in self.libraries]
```

Comment:
I think I would prefer `[lib.static if isinstance(lib, BothLibraries) else lib for lib in self.libraries]`. Same pattern as you did for ext_deps. Just in case other types could have a `static`  attribute with a different meaning.

Reply:
I found the former shorter and more clever... but you are probably right.